### PR TITLE
add delete option in product-type.

### PR DESCRIPTION
**Pull request by dr3dd589**
added an option in product type so that we can also delete it. And also beutify the '/product_type/views.py' file according to flake8.

- [x] Your code is flake8 compliant (DefectDojo's code isn't currently flake8 compliant, but we're trying to correct that.)
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [ ] Add applicable tests to the unit tests.

### CODE REVIEW

**Comment by dr3dd589** [comment]
hello @devGregA  Can you please review this now . Let me know if anything have to change. :smiley:

File: dojo/product_type/views.py [hunk 1]
```diff
@@ -22,70 +24,99 @@
...
-    if request.method == 'POST':
-        form = Product_TypeForm(request.POST, instance=pt)
-        if form.is_valid():
-            pt = form.save()
-            messages.add_message(request,
-                                 messages.SUCCESS,
-                                 'Product type updated successfully.',
-                                 extra_tags='alert-success')
-            return HttpResponseRedirect(reverse('product_type'))
+    form1 = Product_TypeForm(instance=pt)
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]: Use meaningful names for variables, avoiding names such as `form1`, `form2`, etc.
```suggestion
    pt_form = Product_TypeForm(instance=pt)
```

File: dojo/product_type/views.py [hunk 2]
```diff
@@ -22,70 +24,99 @@
 
 
 def product_type(request):
-    initial_queryset = Product_Type.objects.all().order_by('name')
-    name_words = [product.name for product in
-                  initial_queryset]
+    initial_queryset = Product_Type.objects.all().order_by("name")
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]: As recommended by PEP- 498, let's keep single quotes, unless double quotes are required.
```suggestion
    initial_queryset = Product_Type.objects.all().order_by('name')
```

File: dojo/product_type/views.py [hunk 3]
```diff
@@ -22,70 +24,99 @@
...
-    return render(request, 'dojo/product_type.html', {
-        'name': 'Product Type List',
-        'metric': False,
-        'user': request.user,
-        'pts': pts,
-        'ptl': ptl,
-        'name_words': name_words})
+    return render(
+        request,
+        "dojo/product_type.html",
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]: As recommended by [PEP- 498](https://www.python.org/dev/peps/pep-0498/), let's keep single quotes, unless double quotes are required.
```suggestion
        'dojo/product_type.html',
```
And following lines.

File: dojo/product_type/views.py [hunk 4]
```diff
@@ -22,70 +24,99 @@
...
+            "name_words": name_words,
+        },
+    )
 
 
 @user_passes_test(lambda u: u.is_staff)
 def add_product_type(request):
     form = Product_TypeForm()
-    if request.method == 'POST':
+    if request.method == "POST":
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]: As recommended by [PEP- 498](https://www.python.org/dev/peps/pep-0498/), let's keep single quotes, unless double quotes are required.
```suggestion
    if request.method == 'POST':
```

File: dojo/product_type/views.py [hunk 5]
```diff
@@ -22,70 +24,99 @@
...
             form.save()
-            messages.add_message(request,
-                                 messages.SUCCESS,
-                                 'Product type added successfully.',
-                                 extra_tags='alert-success')
-            return HttpResponseRedirect(reverse('product_type'))
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                "Product type added successfully.",
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]: As recommended by [PEP- 498](https://www.python.org/dev/peps/pep-0498/), let's keep single quotes, unless double quotes are required.
```suggestion
                'Product type added successfully.',
```
And following lines.

File: dojo/product_type/views.py [hunk 6]
```diff
@@ -22,70 +24,99 @@
...
     add_breadcrumb(title="Add Product Type", top_level=False, request=request)
-    return render(request, 'dojo/new_product_type.html', {
-        'name': 'Add Product Type',
-        'metric': False,
-        'user': request.user,
-        'form': form,
-    })
+    return render(
+        request,
+        "dojo/new_product_type.html",
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]: As recommended by [PEP- 498](https://www.python.org/dev/peps/pep-0498/), let's keep single quotes, unless double quotes are required.
```suggestion
        'dojo/new_product_type.html',
```
And following lines.

File: dojo/product_type/views.py [hunk 7]
```diff
@@ -22,70 +24,99 @@
...
-        form = Product_TypeForm(request.POST, instance=pt)
-        if form.is_valid():
-            pt = form.save()
-            messages.add_message(request,
-                                 messages.SUCCESS,
-                                 'Product type updated successfully.',
-                                 extra_tags='alert-success')
-            return HttpResponseRedirect(reverse('product_type'))
+    form1 = Product_TypeForm(instance=pt)
+    form2 = Delete_Product_TypeForm(instance=pt)
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]: Use meaningful names for variables, avoiding names such as `form1`, `form2`, etc.
```suggestion
    delete_pt_form = Delete_Product_TypeForm(instance=pt)
```

File: dojo/product_type/views.py [hunk 8]
```diff
@@ -22,70 +24,99 @@
...
+    if request.method == "POST" and request.POST.get("edit_product_type"):
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]: As recommended by [PEP- 498](https://www.python.org/dev/peps/pep-0498/), let's keep single quotes, unless double quotes are required.
```suggestion
    if request.method == "POST" and request.POST.get('edit_product_type'):
```

File: dojo/product_type/views.py [hunk 9]
```diff
@@ -22,70 +24,99 @@
...
+        form1 = Product_TypeForm(request.POST, instance=pt)
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]:
```suggestion
        pt_form = Product_TypeForm(request.POST, instance=pt)
```

File: dojo/product_type/views.py [hunk 10]
```diff
@@ -22,70 +24,99 @@
...
+        if form1.is_valid():
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]:
```suggestion
        if pt_form.is_valid():
```

File: dojo/product_type/views.py [hunk 11]
```diff
@@ -22,70 +24,99 @@
...
+            pt = form1.save()
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]:
```suggestion
            pt = pt_form.save()
```

File: dojo/product_type/views.py [hunk 12]
```diff
@@ -22,70 +24,99 @@
...
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                "Product type updated successfully.",
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]: As recommended by [PEP- 498](https://www.python.org/dev/peps/pep-0498/), let's keep single quotes, unless double quotes are required.
```suggestion
                'Product type updated successfully.',
```
And following lines.

File: dojo/product_type/views.py [hunk 13]
```diff
@@ -22,70 +24,99 @@
...
     add_breadcrumb(title="Edit Product Type", top_level=False, request=request)
-    return render(request, 'dojo/edit_product_type.html', {
-        'name': 'Edit Product Type',
-        'metric': False,
-        'user': request.user,
-        'form': form,
-        'pt': pt})
+    return render(
+        request,
+        "dojo/edit_product_type.html",
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]: As recommended by [PEP- 498](https://www.python.org/dev/peps/pep-0498/), let's keep single quotes, unless double quotes are required.
```suggestion
        'dojo/edit_product_type.html',
```
And following lines.

File: dojo/product_type/views.py [hunk 14]
```diff
@@ -22,70 +24,99 @@
...
 
 @user_passes_test(lambda u: u.is_staff)
 def add_product_to_product_type(request, ptid):
     pt = get_object_or_404(Product_Type, pk=ptid)
-    form = Product_TypeProductForm(initial={'prod_type': pt})
-    add_breadcrumb(title="New %s Product" % pt.name, top_level=False, request=request)
-    return render(request, 'dojo/new_product.html',
-                  {'form': form,
-                   })
+    form = Product_TypeProductForm(initial={"prod_type": pt})
```
Comment (ppiotr3k) [inline dojo/product_type/views.py]: As recommended by [PEP- 498](https://www.python.org/dev/peps/pep-0498/), let's keep single quotes, unless double quotes are required.
```suggestion
    form = Product_TypeProductForm(initial={'prod_type': pt})
```
And following lines.

File: dojo/templates/dojo/edit_product_type.html [hunk 15]
```diff
@@ -2,11 +2,13 @@
 {% block content %}
     <h3> Edit Product Type {{ pt.name }}</h3>
     <form class="form-horizontal" action="{% url 'edit_product_type' pt.id %}" method="post">{% csrf_token %}
-        {% include "dojo/form_fields.html" with form=form %}
+        {% include "dojo/form_fields.html" with form=form1 %}
```
Comment (ppiotr3k) [inline dojo/templates/dojo/edit_product_type.html]:
```suggestion
        {% include "dojo/form_fields.html" with form=pt_form %}
```

File: dojo/templates/dojo/edit_product_type.html [hunk 16]
```diff
@@ -2,11 +2,13 @@
...
+        {{ form2 }}
```
Comment (ppiotr3k) [inline dojo/templates/dojo/edit_product_type.html]:
```suggestion
        {{ delete_pt_form }}
```

File: dojo/templates/dojo/edit_product_type.html [hunk 17]
```diff
@@ -2,11 +2,13 @@
...
         <div class="form-group">
             <div class="col-sm-offset-2 col-sm-10">
-                <input class="btn btn-primary" type="submit" value="Submit"/>
+                <input class="btn btn-primary" name="edit_product_type" type="submit" value="Submit"/>
+                <input class="btn btn-danger" name="delete_product_type" type="submit" value="Delete"/>
             </div>
         </div>
     </form>
-{% endblock %}
+{% endblock %}
```
Comment (ppiotr3k) [inline dojo/templates/dojo/edit_product_type.html]: Newline required at end of file.

**Comment by dr3dd589** [comment]
thanks. @ppiotr3k :smiley:

**Comment by dr3dd589** [comment]
hello, @devGregA  Can you please review this!

**Comment by valentijnscholten** [comment]
I was about to add this feature myself, until I found this PR. So good news, and thanks for the PR.
I do think the PR stil contains too many changes that are just formatting changes and not related to the added functionality. It does look OK but it requires us to mix and match different lines to see what/if there are changes. It would easier to review if the PR contained just the new stuff.

**Comment by dr3dd589** [comment]
hello, @valentijnscholten thanks for reviewing. Can you suggest to me what can I do now so that it would be easier to review in this PR?

**Comment by valentijnscholten** [comment]
Make sure views.py only contains logic changes, so remove the changes that only are modifying whitespace or code formatting. (not sure if the flake8 will complain about that)

**Comment by dr3dd589** [comment]
thanks! done.

**Comment by valentijnscholten** [comment]
Looks good, just checking with @aaronweaver what the policy around the access linting suggestion is?

**Comment by aaronweaver** [comment]
@valentijnscholten It's mostly looking for labels etc to assist in readability with screen readers. I just added it so I'll check and see what other items it checks.

**Comment by valentijnscholten** [comment]
@aaronweaver was just asking to see if all accesslint issues need to be resolved berfore we can approve PRs

**Comment by aaronweaver** [comment]
@valentijnscholten Yes, let's start doing it we have at least one user who will benefit from small changes like this.

**Comment by valentijnscholten** [comment]
ok, good to know. We'll wait then for @dr3dd589 to make the suggested changes.

**Comment by valentijnscholten** [comment]
I would like to approve this, but Github doesn't offer me the "add your review" option. Probably because I already approved it earlier? However when I click "Show all reviewers", it only shows the review of @adracea ...

**Comment by devGregA** [comment]
@valentijnscholten weird. Let me see if I can figure it out

**Comment by devGregA** [comment]
Very strange. It shows you approving in the top right of the page.

**Comment by valentijnscholten** [comment]
It only shows a green tick at Andrea's name, mine is black. Maybe just merge the pr and forget about it 🙂

**Comment by dr3dd589** [comment]
On hovering tick sign it shows:
 `only reviews by reviewers with write access count towards mergeability`

**Comment by valentijnscholten** [comment]
When I hover, it says "valentijnscholten approved these changes". Could it be a bug in github because the PR was created before I got write access?

**Comment by dr3dd589** [comment]
No , just hover on bottom tick where you approved this, not top right corner tick.

**Comment by valentijnscholten** [comment]
Ah yes, still doesn't make sense because I have write access. Must be a github hickup, let's merge :D

**Comment by devGregA** [comment]
Merged :)

**Pull request merged**